### PR TITLE
[PATCH v7] odp-generator: Use clock related functions less when printing

### PR DESCRIPTION
**Pull request by apalos**
Using odp_time_local() on each iteration consumes too much cpu for no reason
Netmap pktio reported ~550kpps prior to this and ~1.2Mpps after the patch, when
odp_generator was used as a receiver.
Signed-off-by: Ilias Apalodimas <ilias.apalodimas@linaro.org>

### CODE REVIEW

File: example/generator/odp_generator.c [hunk 1]
```diff
@@ -765,11 +765,12 @@ static int gen_recv_thread(void *arg)
  */
 static void print_global_stats(int num_workers)
 {
-	odp_time_t cur, wait, next;
+	odp_time_t cur, wait, next, left;
 	uint64_t pkts_snd = 0, pkts_snd_prev = 0;
 	uint64_t pps_snd = 0, maximum_pps_snd = 0;
 	uint64_t pkts_rcv = 0, pkts_rcv_prev = 0;
 	uint64_t pps_rcv = 0, maximum_pps_rcv = 0;
+	uint64_t stall = 0;
```
Comment (Bill-Fischofer-Linaro) [inline example/generator/odp_generator.c]: Nit: Does GCC flag this as an uninitialized variable without the initialization here? That doesn't seem needed.

Comment (apalos) [reply example/generator/odp_generator.c]: No there's no warning if we leave that uninitialized

File: example/generator/odp_generator.c [hunk 2]
```diff
@@ -786,8 +787,14 @@ static void print_global_stats(int num_workers)
 		}
 
 		cur = odp_time_local();
-		if (odp_time_cmp(next, cur) > 0)
+		if (odp_time_cmp(next, cur) > 0) {
+			left = odp_time_diff(next, cur);
+			stall = odp_time_to_ns(left);
+			stall = (stall / 1000000000) + 1;
```
Comment (muvarov) [inline example/generator/odp_generator.c]: ODP_TIME_SEC_IN_NS

File: example/generator/odp_generator.c [hunk 3]
```diff
@@ -786,8 +787,14 @@ static void print_global_stats(int num_workers)
 		}
 
 		cur = odp_time_local();
-		if (odp_time_cmp(next, cur) > 0)
+		if (odp_time_cmp(next, cur) > 0) {
+			left = odp_time_diff(next, cur);
+			stall = odp_time_to_ns(left);
+			stall = (stall / ODP_TIME_SEC_IN_NS) + 1;
+			stall &= 0xFFFFFFFF;
```
[the remark this answers is not in the document]
Comment (apalos) [reply example/generator/odp_generator.c]: Adjusted patch accordingly.

**Comment by bogdanPricope** [comment]
Reviewed-and-tested-by: Bogdan Pricope <bogdan.pricope@linaro.org>

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**